static/app/views/preprod/utils/insightProcessing.ts

```diff
@@ -0,0 +1,279 @@
+import type {
+  AppleInsightResults,
+  FileSavingsResult,
+  FileSavingsResultGroup,
+  FilesInsightResult,
+  GroupsInsightResult,
+  OptimizableImageFile,
+  StripBinaryFileInfo,
+} from 'sentry/views/preprod/types/appSizeTypes';
+
+type FileTypeData =
+  | {fileType: 'optimizable_image'; originalFile: OptimizableImageFile}
+  | {fileType: 'strip_binary'; originalFile: StripBinaryFileInfo}
+  | {fileType: 'regular'; originalFile: FileSavingsResult};
+
+export interface ProcessedInsightFile {
+  data: FileTypeData;
+  path: string;
+  percentage: number;
+  savings: number;
+}
+
+export interface ProcessedInsight {
+  description: string;
+  files: ProcessedInsightFile[];
+  name: string;
+  percentage: number;
+  totalSavings: number;
+}
+
+interface InsightConfig {
+  description: string;
+  key: string;
+  name: string;
+}
+
+const INSIGHT_CONFIGS: InsightConfig[] = [
+  {
+    key: 'image_optimization',
+    name: 'Optimize images',
```

Comment:
Just a TODO ([linear issue](https://linear.app/getsentry/issue/EME-125/ensure-all-strings-are-translatedtransalatable)), we'll need to translate these in the future. Seems a `t()` function kicks that process off. 

Personally I say we do that all in one fell-swoop, esp since this page is in churn, but might be good to leave a todo comment here

Reply:
Oh yea sorry I mentioned this to Jonas offline, going to wait for a final copy review before sending to translators.